pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/results/BaseResultsBlock.java

```diff
@@ -174,6 +177,21 @@ protected void attachMetadataToDataTable(DataTable dataTable) {
     metadata.put(MetadataKey.NUM_CONSUMING_SEGMENTS_PROCESSED.getName(),
         Integer.toString(_numConsumingSegmentsProcessed));
     metadata.put(MetadataKey.NUM_CONSUMING_SEGMENTS_MATCHED.getName(), Integer.toString(_numConsumingSegmentsMatched));
+    return metadata;
+  }
+
+  protected void attachMetadataToDataTable(DataTable dataTable) {
```

Comment:
hmm I wonder if the API has now become slightly less intuitive ?

Initially every concrete block will either override attachMetadataToDataTable and/or call super.attachMetadataToDataTable. 

Now we have this additional API to just compute the metadata ? Why do we need to special case just the metadata compute part ?

Reply:
yes. i wanted to remove this. i am rethinking this is -->
once the BaseResultBlock gets wrapped into an InstanceResponseBlock. everything other than the real data should be move into the instanceResponseBlock's member variable. and this should remain unchanged.

where as the metadata in InstanceResponseBlock will get enriched with downstream logic. (metrics, error, warnings, etc)